### ISSUE TITLE
Properly hide or disable the field

### ISSUE THREAD

**Issue by JanStevens** [voorhoede/datocms-plugin-computed-fields#20]
Hi,

I noticed that the computed field is non editable but not indicated to the editor. DatoCMS supports marking the field as disabled.

Same for hiding the field, in DatoCMS its possible to remove the field completely by not rendering anything.

Example from a plugin I once made

https://github.com/novemberfiveco/datocms-plugin-nanoid/blob/fba8a22bbf336c37995cd47c741702f0996d0a28/src/entrypoints/FieldExtension.tsx#L29

Would you be open to accept a PR for this?

Regards

**Comment by Siilwyn** [comment]
Hi @JanStevens thank you for your contribution, we'll get back to you :)
Hiding is already a feature right? Do you mean not rendering the title too?

@velomovies thoughts on rendering the field as disabled?
Can imagine not rendering the title could be nice but also could lead to confusion where data is coming from?

**Comment by JanStevens** [comment]
@Siilwyn I indeed mean not rendering anything at all (see related issue: https://github.com/voorhoede/datocms-plugin-computed-fields/issues/19#issuecomment-1335269048)

At least we should render the field disabled to clearly indicated to content editors this is an auto generated field

**Comment by velomovies** [comment]
Hi @JanStevens, 

For hiding the field we already returned nothing in the previous version. I've changed that to returning `null` instead of a `Fragment`. 

In your example you use the `hideField` function from Dato next to returning `null`. That works great in your example, but not in this case. The `hideField` function will not render the form input, and makes it so the plugin doesn't rerender when something changes in the other fields. Therefore the value of the computed field will not update automatically when you change other fields in the model. The automatic update is something we very heavily rely on and don't want to break when hiding a field.

Showing the field as `disabled` is a great addition and will be implemented when we update the plugin.
